Scary messages on normal startup

### FastRPC version

v1.0.2

### DSP firmware version

QC_IMAGE_VERSION_STRING=DSP.AT.1.0.1-00170-LEMANS-1

### SoC and device name

Monaco EVK (RB4) QCS8275 (QCS8300)

### Kernel version

6.19.0-g1cfc3259c7dc

### What happened?

After installing fastrpc and related packages (hexagon-dsp-binaries), a normal startup shows a lot of errors in the journal

I actually think these errors are benign, but:
- they are scary for developers
- they participate in hiding real issues that might pop up


<img width="1731" height="980" alt="Image" src="https://github.com/user-attachments/assets/4689f992-7e1e-4a41-a98c-3af01b67c190" />

### Steps to reproduce

Install fastrpc packages and firmware dependencies, reboot, check journalctl



### Relevant log output

```shell

```

Some of the errors are resolved by PR #299. Checking the remaining error logs.

```
root@qcs8300-ride-sx:/# journalctl | grep adsprpcd
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: dsprpcd.c:174:ADSP daemon starting
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: dsprpcd.c:174:ADSP daemon starting
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: fastrpc_config_parser.c:93: dsp_lib_paths is qcs8300/Qualcomm/QCS8300-RIDE/dsp
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: fastrpc_config_parser.c:93: dsp_lib_paths is qcs8300/Qualcomm/QCS8300-RIDE/dsp
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: fastrpc_apps_user.c:3979: fastrpc_apps_user_init done. default domain:0 and &fastrpc_trace:0xffffae258568
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: fastrpc_apps_user.c:3979: fastrpc_apps_user_init done. default domain:0 and &fastrpc_trace:0xffffb9968568
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: fastrpc_apps_user.c:4100: multidsplib_env_init: libadsprpc.so loaded
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: fastrpc_apps_user.c:4100: multidsplib_env_init: libadsprpc.so loaded
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: dsprpcd.c:181:adsp_default_listener_start called
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: dsprpcd.c:181:adsp_default_listener_start called
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: fastrpc_apps_user.c:2862: Invalid domain name: audiopd
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: fastrpc_apps_user.c:1672: remote_handle_open: Successfully opened handle 0x103 for '":;./\attachguestos on domain 0 (spawn time 0 us, load time 0 us)
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: fastrpc_apps_user.c:1672: remote_handle_open: Successfully opened handle 0x101 for '":;./\createstaticpd:audiopd on domain 0 (spawn time 0 us, load time 0 us)
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: fastrpc_perf.c:323: fastrpc_perf_init: enabled systrace 0x0 and RPC traces (kernel 0, dsp 0) with frequency 1000
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: log_config.c:397: file_watcher_thread starting for domain 0
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: log_config.c:421:file_watcher_thread: Couldn't find file adsprpcd.farf, errno (No such file or directory) at /usr/share/qcom//qcs8300/Qualcomm/QCS8300-RIDE/dsp;/usr/lib/rfsa/adsp;/usr/lib/dsp;
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: fastrpc_perf.c:323: fastrpc_perf_init: enabled systrace 0x0 and RPC traces (kernel 0, dsp 0) with frequency 1000
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: log_config.c:397: file_watcher_thread starting for domain 0
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: log_config.c:421:file_watcher_thread: Couldn't find file adsprpcd.farf, errno (No such file or directory) at /usr/share/qcom//qcs8300/Qualcomm/QCS8300-RIDE/dsp;/usr/lib/rfsa/adsp;/usr/lib/dsp;
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: fastrpc_apps_user.c:1491: Error 0x8000043e: remote_handle64_invoke failed for module (null), handle 0xffffac000be0, method 5 on domain 0 (sc 0x5000000) (errno Success)
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: listener_android.c:352: Error 0x8000043e: listener_start_thread domains support not available in listener
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: listener_android.c:131: listener thread starting
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: fastrpc_apps_user.c:1491: Error 0x8000043e: remote_handle64_invoke failed for module (null), handle 0xaaaadd0f9d70, method 2 on domain 0 (sc 0x2020200) (errno Success)
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: fastrpc_apps_user.c:1599: Warning 0x8000043e: remote_handle_open_domain: remotectl1 domains not supported for domain 0
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: fastrpc_apps_user.c:1491: Error 0x8000043e: remote_handle64_invoke failed for module (null), handle 0xffffa0000be0, method 5 on domain 0 (sc 0x5000000) (errno Success)
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: listener_android.c:352: Error 0x8000043e: listener_start_thread domains support not available in listener
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: listener_android.c:131: listener thread starting
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: fastrpc_apps_user.c:1672: remote_handle_open: Successfully opened handle 0xb033d050 for adsp_default_listener on domain 0 (spawn time 4467 us, load time 2240 us)
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: fastrpc_apps_user.c:1672: remote_handle_open: Successfully opened handle 0xb033d5f0 for adsp_default_listener on domain 0 (spawn time 0 us, load time 230 us)
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: fastrpc_apps_user.c:1672: remote_handle_open: Successfully opened handle 0x8 for '":;./\geteventfd on domain 0 (spawn time 0 us, load time 0 us)
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[654]: mod_table.c:909: Error 0x2: open_mod_table_handle_invoke failed for handle:0xae25c188, sc:0x13050100
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: fastrpc_apps_user.c:1491: Error 0x8000043e: remote_handle64_invoke failed for module (null), handle 0xaaaac3e444e0, method 2 on domain 0 (sc 0x2020200) (errno Success)
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: fastrpc_apps_user.c:1599: Warning 0x8000043e: remote_handle_open_domain: remotectl1 domains not supported for domain 0
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: fastrpc_apps_user.c:1672: remote_handle_open: Successfully opened handle 0xf059cf80 for adsp_default_listener on domain 0 (spawn time 13172 us, load time 3699 us)
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: fastrpc_apps_user.c:1672: remote_handle_open: Successfully opened handle 0xf059d2b0 for adsp_default_listener on domain 0 (spawn time 0 us, load time 1743 us)
Jan 02 18:21:10 qcs8300-ride-sx adsprpcd[653]: fastrpc_apps_user.c:1672: remote_handle_open: Successfully opened handle 0x8 for '":;./\geteventfd on domain 0 (spawn time 0 us, load time 0 us)
```

```
root@qcs8300-ride-sx:/# journalctl | grep cdsprpcd
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: dsprpcd.c:174:CDSP daemon starting
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: fastrpc_config_parser.c:93: dsp_lib_paths is qcs8300/Qualcomm/QCS8300-RIDE/dsp
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: fastrpc_apps_user.c:3979: fastrpc_apps_user_init done. default domain:3 and &fastrpc_trace:0xffff9e957568
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: fastrpc_apps_user.c:4100: multidsplib_env_init: libcdsprpc.so loaded
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: dsprpcd.c:181:adsp_default_listener_start called
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: fastrpc_apps_user.c:1672: remote_handle_open: Successfully opened handle 0x103 for '":;./\attachguestos on domain 3 (spawn time 0 us, load time 0 us)
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: fastrpc_perf.c:323: fastrpc_perf_init: enabled systrace 0x0 and RPC traces (kernel 0, dsp 0) with frequency 1000
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: log_config.c:397: file_watcher_thread starting for domain 3
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: log_config.c:421:file_watcher_thread: Couldn't find file cdsprpcd.farf, errno (No such file or directory) at /usr/share/qcom//qcs8300/Qualcomm/QCS8300-RIDE/dsp;/usr/lib/rfsa/adsp;/usr/lib/dsp;
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: fastrpc_apps_user.c:1491: Error 0x8000043e: remote_handle64_invoke failed for module (null), handle 0xffff98000be0, method 5 on domain 3 (sc 0x5000000) (errno Success)
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: listener_android.c:352: Error 0x8000043e: listener_start_thread domains support not available in listener
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: listener_android.c:131: listener thread starting
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: fastrpc_apps_user.c:1491: Error 0x8000043e: remote_handle64_invoke failed for module (null), handle 0xaaaad8887520, method 2 on domain 3 (sc 0x2020200) (errno Success)
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: fastrpc_apps_user.c:1599: Warning 0x8000043e: remote_handle_open_domain: remotectl1 domains not supported for domain 3
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: fastrpc_apps_user.c:1672: remote_handle_open: Successfully opened handle 0xf03e8880 for adsp_default_listener on domain 3 (spawn time 12237 us, load time 1540 us)
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: fastrpc_apps_user.c:1672: remote_handle_open: Successfully opened handle 0xf03e8ae0 for adsp_default_listener on domain 3 (spawn time 1 us, load time 799 us)
Jan 02 18:21:10 qcs8300-ride-sx cdsprpcd[669]: fastrpc_apps_user.c:1672: remote_handle_open: Successfully opened handle 0x8 for '":;./\geteventfd on domain 3 (spawn time 0 us, load time 0 us)
```

--- a/src/fastrpc_apps_user.c
+++ b/src/fastrpc_apps_user.c
@@ -1488,11 +1488,19 @@ bail:
     }
     if (0 == check_rpc_error(nErr)) {
       if (get_logger_state(domain)) {
-        FARF(ERROR,
-             "Error 0x%x: %s failed for module %s, handle 0x%" PRIx64
-             ", method %d on domain %d (sc 0x%x) (errno %s)\n",
-             nErr, __func__, h->name, local, REMOTE_SCALARS_METHOD(sc), domain, sc,
-             strerror(errno));
+        if (nErr == DSP_AEE_EOFFSET + AEE_ENOSUCHMOD) {
+          FARF(ALWAYS,
+               "Warning: %s module %s is not supported on DSP domain %d, handle 0x%" PRIx64
+               ", method %d (sc 0x%x) (errno %s)\n",
+               __func__, h->name, domain, local, REMOTE_SCALARS_METHOD(sc), sc,
+               strerror(errno));
+        } else {
+          FARF(ERROR,
+               "Error 0x%x: %s failed for module %s, handle 0x%" PRIx64
+               ", method %d on domain %d (sc 0x%x) (errno %s)\n",
+               nErr, __func__, h->name, local, REMOTE_SCALARS_METHOD(sc), domain, sc,
+               strerror(errno));
+        }
       }
     }
   }

--- a/src/listener_android.c
+++ b/src/listener_android.c
@@ -349,8 +349,9 @@ static void *listener_start_thread(void *arg) {
     nErr = __QAIC_HEADER(adsp_listener1_init2)(adsp_listener1_handle);
     if ((nErr == DSP_AEE_EOFFSET + AEE_ERPC) ||
         nErr == DSP_AEE_EOFFSET + AEE_ENOSUCHMOD) {
-      FARF(ERROR, "Error 0x%x: %s domains support not available in listener",
-           nErr, __func__);
+      FARF(ALWAYS,
+           "Warning: %s domain support is unavailable on DSP in listener",
+           __func__);
       fastrpc_update_module_list(DOMAIN_LIST_DEQUEUE, domain, _const_adsp_listener1_handle, NULL, NULL);
       adsp_listener1_handle = INVALID_HANDLE;
       VERIFY(AEE_SUCCESS == (nErr = __QAIC_HEADER(adsp_listener_init2)()));

